openssl/src/bn/mod.rs

```diff
@@ -185,53 +185,54 @@ impl<'a> BigNumRef<'a> {
         }
     }
 
-    /// Add an `unsigned long` to `self`. This is more efficient than adding a `BigNum`.
-    pub fn add_word(&mut self, w: c_ulong) -> Result<(), ErrorStack> {
+    /// Add a `u32` to `self`. This is more efficient than adding a
+    /// `BigNum`.
+    pub fn add_word(&mut self, w: u32) -> Result<(), ErrorStack> {
         unsafe {
-            if ffi::BN_add_word(self.as_ptr(), w) == 1 {
+            if ffi::BN_add_word(self.as_ptr(), w as ffi::BN_ULONG) == 1 {
                 Ok(())
             } else {
                 Err(ErrorStack::get())
             }
         }
     }
 
-    pub fn sub_word(&mut self, w: c_ulong) -> Result<(), ErrorStack> {
+    pub fn sub_word(&mut self, w: u32) -> Result<(), ErrorStack> {
```

Comment:
This was actually an incorrect definition of the `BN_ULONG` type alias. (evidenced by the changes in the sys crate above). The definition of this is `c_uint` on Windows which is only 32-bits.

I figured it's safe at least to consume the *smallest* bit width used across platforms and then we'd return the largest bit width any platform returns. In that sense there shouldn't be a downcast?